src/core/agent/services/signifyNotificationService.ts

```diff
@@ -218,6 +226,50 @@ class SignifyNotificationService extends AgentService {
     });
     return notificationRecord;
   }
+  async onSignifyOperationStateChanged(
+    callback: (identifierShortDetails: IdentifierShortDetails) => void
+  ) {
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const pendingIdentifiers =
+        await this.identifierStorage.getAllPendingIdentifierMetadata();
```

Comment:
I think I'd prefer if we created a new record for pending operations to track them that way, instead of via pending identifiers. Because we will use this for other things so if it was generic it would be better.

Reply:
We should also fetch the pending operations once on start-up and add/remove dynamically rather than fetching from the DB each time

Reply:
Can you check new code?. I think we need get from db to check and update/delete it.